stix_shifter/stix_transmission/src/modules/utils/RestApiClient.py

```diff
@@ -47,24 +77,34 @@ def call_api(self, endpoint, method, headers=None, params=[], data=None, urldata
                 endpoint += urldata
 
             if self.url_modifier_function is not None:
-                url = self.url_modifier_function(self.server_ip, endpoint, actual_headers)
+                url = self.url_modifier_function(
+                    self.server_ip, endpoint, actual_headers)
             else:
                 url = 'https://' + self.server_ip + '/' + endpoint
             try:
                 call = getattr(requests, method.lower())
-                response = call(url, headers=actual_headers, cert=self.cert_file_name, data=data, verify=self.cert_verify)
-                
+
+                response = call(url, headers=actual_headers,
+                                cert=self.client_cert_content, data=data, verify=self.server_cert_content)
+
                 if 'headers' in dir(response) and isinstance(response.headers, collections.Mapping) and 'Content-Type' in response.headers \
-                                and "Deprecated" in response.headers['Content-Type']:
-                    print("WARNING: " + response.headers['Content-Type'], file=sys.stderr)
+                        and "Deprecated" in response.headers['Content-Type']:
+                    print("WARNING: " +
+                          response.headers['Content-Type'], file=sys.stderr)
                 return ResponseWrapper(response)
             except Exception as e:
                 print('exception occured during requesting url: ' + str(e))
                 raise e
         finally:
-            if self.cert_file_name is not None:
+            if self.server_cert_exist:
```

Comment:
Can't you just check `self.server_cert_content` rather than creating this extra variable?

Reply:
Yea for sure